drivers/android/process.rs

```diff
@@ -293,22 +293,24 @@ unsafe impl Sync for Process {}
 
 impl Process {
     fn new(ctx: Arc<Context>) -> Result<Ref<Self>> {
-        let mut proc_ref = Ref::try_new(Self {
-            ref_count: RefCount::new(),
-            ctx,
-            // SAFETY: `inner` is initialised in the call to `mutex_init` below.
-            inner: unsafe { Mutex::new(ProcessInner::new()) },
-            // SAFETY: `node_refs` is initialised in the call to `mutex_init` below.
-            node_refs: unsafe { Mutex::new(ProcessNodeRefs::new()) },
-        })?;
-        let process = Ref::get_mut(&mut proc_ref).ok_or(Error::EINVAL)?;
-        // SAFETY: `inner` is pinned behind the `Arc` reference.
-        let pinned = unsafe { Pin::new_unchecked(&process.inner) };
-        kernel::mutex_init!(pinned, "Process::inner");
-        // SAFETY: `node_refs` is pinned behind the `Arc` reference.
-        let pinned = unsafe { Pin::new_unchecked(&process.node_refs) };
-        kernel::mutex_init!(pinned, "Process::node_refs");
-        Ok(proc_ref)
+        Ref::try_new_and_init(
+            Self {
+                ref_count: RefCount::default(),
+                ctx,
+                // SAFETY: `inner` is initialised in the call to `mutex_init` below.
+                inner: unsafe { Mutex::new(ProcessInner::new()) },
+                // SAFETY: `node_refs` is initialised in the call to `mutex_init` below.
+                node_refs: unsafe { Mutex::new(ProcessNodeRefs::new()) },
+            },
+            |process| {
+                // SAFETY: `inner` is pinned behind the `Ref` reference.
+                let pinned = unsafe { Pin::new_unchecked(&process.inner) };
+                kernel::mutex_init!(pinned, "Process::inner");
+                // SAFETY: `node_refs` is pinned behind the `Ref` reference.
+                let pinned = unsafe { Pin::new_unchecked(&process.node_refs) };
+                kernel::mutex_init!(pinned, "Process::node_refs");
+            },
```

Comment:
See https://github.com/wedsonaf/linux/commit/61397b0c79a775f12228e47ed1ab47839d591531 -- this allows us to obviate `get_mut`, which allows us to safely claim that `Ref` is in fact pinned.

Reply:
See https://github.com/wedsonaf/linux/commit/e201bf1b9f5db58e5b5608d88b924d8ad5cdc14e -- allowing `init` on just `&self` (as opposed to `&mut self`) was an oversight that leads to unsafety (e.g., someone calling `init` on a synchronisation primitive that is in use).

Reply:
Holy smoke! I was already wondering about that, but I'm coming from profound ignorance on `Pin`ning and everything around it.

Reply:
The best way to avoid `get_mut` is through a wrapper type like `UniqueArc` https://doc.servo.org/servo_arc/struct.UniqueArc.html

Reply:
> allowing `init` on just `&self` (as opposed to `&mut self`) was an oversight that leads to unsafety

PS do we have an Issue to keep track of this?

Reply:
> The best way to avoid `get_mut` is through a wrapper type like `UniqueArc` https://doc.servo.org/servo_arc/struct.UniqueArc.html

Nice, I like it.

Something for another PR though. (The reason I'm removing `get_mut` in this PR is that we don't have a `refcount_t` function with the right barriers for `get_mut` and I want this code to be simple wrappers around `refcount_t`).

> > allowing `init` on just `&self` (as opposed to `&mut self`) was an oversight that leads to unsafety
> 
> PS do we have an Issue to keep track of this?

I don't think so. Feel free to open one.